lib/elixir/lib/io.ex

```diff
@@ -171,14 +178,13 @@ defmodule IO do
   end
 
   @doc """
-  Writes the argument to the device, similar to `write/2`,
-  but adds a newline at the end. The argument is expected
-  to be a chardata.
+  Writes the given argument to the `device`, similar to `write/2`,
+  but adds a newline at the end.
+
   """
   @spec puts(device, chardata | String.Chars.t) :: :ok
   def puts(device \\ group_leader(), item) do
-    erl_dev = map_dev(device)
```

Comment:
Everywhere else in the code map_dev/1 is used directly. you can see in the function above.